python-threatexchange/threatexchange/threat_updates.py

```diff
@@ -117,12 +117,14 @@ def __init__(
         privacy_group: int,
         start: int = 0,
         end: t.Optional[int] = None,
+        types: t.Iterable[str] = (),
     ) -> None:
         self.privacy_group = privacy_group
         self.updates = []
         self.current = start
         self.start = start
         self.end = end
+        self.types = types
```

Comment:
blocking: This is super unsafe because you accept and store the input iterable, which is unchanged.

This will cause really hard to debug behavior if you pass in a generator
```
threat_updates(a, b, c, {1:"a", 3:"b"}.values())

for t in threat_updates.types:
  print(t)  # prints "a b"

for t in threat_updates.types:
  print(t)  # prints nothing

```

Reply:
changed to list(types)